### PR TITLE
improve: use `serde with`

### PR DESCRIPTION
**Pull request by thesimplekid**
Thanks for pointing me towards this in https://github.com/elsirion/clnurl/pull/8#pullrequestreview-1356381540.  I wonder if it should be serialized this way upstream https://github.com/ElementsProject/lightning/pull/5306?

### CODE REVIEW

File: src/serde_amount.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//! Serde serialization for amount as msat
+
+use cln_rpc::primitives::Amount;
+
+pub trait SerdeAmount: Copy + Sized {
+    fn from_msat(msats: u64) -> Self;
+    fn as_msat(&self) -> u64;
+}
```
Comment (elsirion) [inline src/serde_amount.rs:8]: Was there any particular reason for making the impl generic over this trait?

File: src/main.rs [hunk 2]
```diff
@@ -131,8 +135,8 @@ async fn get_lnurl_struct(
     State(state): State<ClnurlState>,
 ) -> Result<Json<LnurlResponse>, StatusCode> {
     Ok(Json(LnurlResponse {
-        min_sendable: AmountWrapper::from_msat(1),
-        max_sendable: AmountWrapper::from_msat(100000000000),
+        min_sendable: Amount::from_msat(100),
```
Comment (elsirion) [inline src/main.rs:138]: Should probably be configurable instead of tinkering with it in the code.

**Review by elsirion — APPROVED** [review]
I think during the transitioning period deserialization should work for both integers and strings with the suffix while serialization should always write the integer, much like was done here in @laanwj's CLN RPC crate https://github.com/laanwj/cln4rust/pull/53.

EDIT: Lmk if you decide to upstream then we can iterate on a library-quality solution together  :)

That being said, I only target the latest release, so we should be good with only supporting the version without suffix.

**Comment by elsirion** [comment]
Also thx for continuing to work on it, still need to redeploy to test it out.

File: src/serde_amount.rs (hunk 1 above)
Comment (thesimplekid) [reply src/serde_amount.rs:8]: No good reason, just what I got to work looking at `rust-bitcoin`

Comment (thesimplekid) [reply src/serde_amount.rs:8]: Removed the trait in number #11. As its not necessary as we only need the one unit unlike `rust-bitcoin`? Do let me know if that's not what you meant

File: src/main.rs (hunk 2 above)
Comment (thesimplekid) [reply src/main.rs:138]: #10

File: src/serde_amount.rs (hunk 1 above)
Comment (elsirion) [reply src/serde_amount.rs:8]: Yep, that was exactly my point :)

**Pull request merged**